### PR TITLE
tongsuo NTLS client wrapper

### PR DESCRIPTION
**Pull request by itomsawyer**
Test with Tongsuo 8.3.2

Support NTLS "ECC-SM2-WITH-SM4-SM3" client

### CODE REVIEW

File: ntls_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	cipher, err := conn.CurrentCipher()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	t.Log("current cipher", cipher)
+
+	request := `GET / HTTP/1.1
+Host: demo.gmssl.cn
```
Comment (dongbeiouba) [inline ntls_test.go:39]: 请用Tongsuo构建的s_server来起一个服务端

File: ntls_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+User-Agent: tongsuog
```
Comment (dongbeiouba) [inline ntls_test.go:40]: tongsuog -> tongsuogo或者tongsuo-go-sdk

**Review by dongbeiouba — CHANGES_REQUESTED** [review]

File: ntls_test.go (hunk 2 above)
Comment (itomsawyer) [reply ntls_test.go]: Removed HTTP like request , use simple echo request instead

File: ntls_test.go (hunk 1 above)
Comment (itomsawyer) [reply ntls_test.go]: Use golang listener to wrapper a Tongsuo NTLS server

**Review by dongbeiouba — APPROVED** [review]
LGTM

**Pull request merged**